truncate_client/src/lil_bits/battle.rs

```diff
@@ -17,69 +22,161 @@ impl<'a> BattleUI<'a> {
+
+    ui.allocate_rect(battle_rect, Sense::click())
 }
 
 impl<'a> BattleUI<'a> {
     pub fn render(self, ctx: &GameCtx, ui: &mut egui::Ui) {
         let mut theme = ctx.theme.rescale(0.5);
         theme.tile_margin = 0.0;
 
+        let mut battle_rect = Rect::NOTHING;
+
         ui.allocate_ui_with_layout(
             vec2(ui.available_size_before_wrap().x, 0.0),
-            Layout::left_to_right(Align::BOTTOM).with_main_wrap(true),
+            Layout::left_to_right(Align::Center).with_main_wrap(true),
             |ui| {
-                for battle_word in &self.battle.attackers {
-                    render_word(battle_word, ctx, ui);
-                }
-
-                match self.battle.outcome {
-                    truncate_core::judge::Outcome::AttackerWins(_) => {
-                        ui.label(RichText::new("Beats").color(Color32::WHITE));
-                    }
-                    truncate_core::judge::Outcome::DefenderWins => {
-                        ui.label(RichText::new("Loses to").color(Color32::WHITE));
-                    }
-                }
-
-                for battle_word in &self.battle.defenders {
-                    render_word(battle_word, ctx, ui);
-                }
+                let words = get_galleys(&self.battle.attackers, ctx, ui);
+                battle_rect = battle_rect.union(paint_galleys(words, ui, false).rect);
             },
         );
+        ui.add_space(5.0);
+
+        let (msg, border_color) = match self.battle.outcome {
+            truncate_core::judge::Outcome::AttackerWins(_) => {
+                ("won an attack against", ctx.theme.addition.darken())
+            }
+            truncate_core::judge::Outcome::DefenderWins => {
+                ("failed an attack against", ctx.theme.defeated.darken())
+            }
+        };
+        let galley = ui.painter().layout_no_wrap(
+            msg.to_string(),
+            FontId::new(
+                ctx.theme.letter_size * 0.3,
+                egui::FontFamily::Name("Truncate-Heavy".into()),
+            ),
+            ctx.theme.text,
+        );
+        battle_rect = battle_rect.union(paint_galleys(vec![galley], ui, false).rect);
+        ui.add_space(5.0);
+
+        ui.allocate_ui_with_layout(
+            vec2(ui.available_size_before_wrap().x, 0.0),
+            Layout::left_to_right(Align::Center).with_main_wrap(true),
+            |ui| {
+                let words = get_galleys(&self.battle.defenders, ctx, ui);
+                battle_rect = battle_rect.union(paint_galleys(words, ui, false).rect);
+            },
+        );
+
+        let battle_interact = ui.interact(battle_rect, ui.auto_id_with("battle"), Sense::click());
+        if battle_interact.hovered() {
+            ui.painter().rect_stroke(
+                battle_rect.expand2(vec2(0.0, 4.0)),
+                2.0,
+                Stroke::new(2.0, border_color),
+            );
+            ui.output_mut(|o| o.cursor_icon = CursorIcon::PointingHand);
```

Comment:
PointingHand implies clicking does something right? And we'll have a dialog box explaining the battle in more detail, but it doesn't do that yet?

Reply:
Yeah for now we're just being duplicitous